tests/functional/services/policy_engine/test_feed_sync.py

```diff
@@ -0,0 +1,104 @@
+import os
+
+import tests.functional.services.policy_engine.utils.api as policy_engine_api
+from tests.functional.services.policy_engine.conftest import FEEDS_DATA_PATH_PREFIX
+from tests.functional.services.utils import http_utils
+
+# TODO check schema
+
+
+class TestFeedSync:
+    @classmethod
+    def _find_by_name(cls, records, name):
+        """
+        From a list of objects/dictionaries, selects first index with matching name. Returns None if nothing is found
+        :param records: list of objects or dictionaries that are expected to have 'name' attr
+        :type records: list
+        :return: dict with matching name or 'None' if nothing found
+        :rtype: Union[dict, None]
+        """
+        for record in records:
+            if record["name"] == name:
+                return record
+        return None
+
+    @classmethod
+    def _get_vuln_ids(cls, expected_vulns):
+        """
+        From a list of expected vulns taken from feeds service, find corresponding vulnerability ids for querying anchore
+        :param expected_vulns: list of records in group from feed
+        :type expected_vulns: list
+        :return: list of vulnerabilitiy_ids as they would be stored in anchore
+        :rtype: list
+        """
+        vuln_ids = []
+        for vuln in expected_vulns:
+            if "Advisory" in vuln:
+                if "ghsaId" in vuln["Advisory"]:
+                    vuln_ids.append(vuln["Advisory"]["ghsaId"])
+                else:
+                    vuln_ids += vuln["Advisory"]["CVE"]
+            if "@id" in vuln:
+                vuln_ids.append(vuln["@id"])
+            if "cve" in vuln:
+                vuln_ids.append(vuln["cve"]["CVE_data_meta"]["ID"])
+            if "Vulnerability" in vuln:
+                if "Name" in vuln["Vulnerability"]:
+                    vuln_ids.append(vuln["Vulnerability"]["Name"])
+        return vuln_ids
+
+    def test_expected_feed_sync(self, expected_content, clear_database_temporary):
+        feed_sync_resp = policy_engine_api.feeds.feeds_sync()
+        assert feed_sync_resp == http_utils.APIResponse(200)
+        for feed in feed_sync_resp.body:
+            assert feed["status"] == "success"
+
+        feeds_get_resp = policy_engine_api.feeds.get_feeds(True)
+
+        # get feeds index file
+        expected_feeds = expected_content(
+            os.path.join(FEEDS_DATA_PATH_PREFIX, "index")
+        )["feeds"]
+
+        assert len(feeds_get_resp.body) == len(expected_feeds)
+
+        for expected_feed in expected_feeds:
```

Comment:
Question - Is it worth it to parameterize these expected values so we get a different test execution for each one? That (IMO) makes the tests harder to read, but we will see if multiple of these start failing, instead of only seeing whichever assertion in these nested loops fails first.

Reply:
Depending on how many items are in each of these collections of expected results you are iterating over, that is potentially a lot of assertions skipped if the the very first one fails.

Reply:
would like to discuss over call

Reply:
parameterize with one entry for each group/feed combo and use programmatic param generation